:memo: Add workers and Blackfire configuration to the Laravel guide (#2288)
platformsh/platformsh-docs: docs/src/guides/laravel/deploy/scheduling.md

---
title: "Workers, cron jobs, and task scheduling"
sidebarTitle: "Scheduling tasks"
weight: -100
description: Schedule tasks for your Laravel app.
---

Laravel offers a very convenient and flexible way of scheduling tasks. A large set of [helper functions](https://laravel.com/docs/scheduling#schedule-frequency-options) help you schedule commands and jobs.

Once the scheduled tasks are defined, they need to be effectively executed at the right time and pace.
The recommended way is a cron configuration entry running the `artisan schedule:run` command.

```yaml {location=".platform.app.yaml"}
crons:
    # Run Laravel's scheduler every 5 minutes, which is as often as crons can run on Professional plans.
    scheduler:
        spec: '*/5 * * * *'
        cmd: 'php artisan schedule:run'
```

> The [minimum time between cron jobs](../../../configuration/app/app-reference.md#cron-job-timing) being triggered depends on your plan. Task scheduling may then be contradicted by the cron minimum frequency. Schedules outside the specified cron frequency will be ignored and the related tasks won't be triggered.

Reviewer: ```suggestion
The [minimum time between cron jobs](../../../configuration/app/app-reference.md#cron-job-timing) being triggered depends on your plan. Task scheduling may then be contradicted by the cron minimum frequency. Schedules outside the specified cron frequency are ignored and the related tasks aren't triggered.
```

We try to avoid the future tense: https://github.com/platformsh/platformsh-docs/blob/main/contributing/content-style.md#use-the-present-tense

Author: Awesome @CollierCZ. I discover this guide, and I have to make it mine. So great to have it in place!

---

> title: "Workers, Cron Jobs and Task Scheduling"
> sidebarTitle: "Scheduling Tasks"

Reviewer: ```suggestion
title: "Workers, cron jobs, and task scheduling"
sidebarTitle: "Scheduling tasks"
```

weight: -100

> description: How to schedule Tasks for your Laravel application.

Reviewer: ```suggestion
description: Schedule tasks for your Laravel app.
```

---

> Laravel offers a very convenient and flexible way of scheduling tasks. A large set of [helpers functions](https://laravel.com/docs/scheduling#schedule-frequency-options) allows the easy scheduling of Commands and Jobs.

Reviewer: ```suggestion
Laravel offers a very convenient and flexible way of scheduling tasks. A large set of [helper functions](https://laravel.com/docs/scheduling#schedule-frequency-options) help you schedule commands and jobs.
```

Once the scheduled tasks are defined, they need to be effectively executed at the right time and pace.
The recommended way is a cron configuration entry running the `artisan schedule:run` command.

```yaml {location=".platform.app.yaml"}
crons:

>     # Run Laravel's scheduler every 5 minutes, which is as often as crons can run.

Reviewer: This is true for Professional plans, but not Elite or Enterprise, as noted immediately below. So I would probably adjust the comment.

    scheduler:
        spec: '*/5 * * * *'
        cmd: 'php artisan schedule:run'
```

> The [minimum time between cron jobs](../../../configuration/app/app-reference.html#cron-job-timing) being triggered depends on your plan. The tasks scheduling may then be contradicted by the cron minimum frequency.

Reviewer: ```suggestion
The [minimum time between cron jobs](../../../configuration/app/app-reference.html#cron-job-timing) being triggered depends on your plan. Task scheduling may then be contradicted by the cron minimum frequency.
```

For the second sentence, what's the result of this contradiction? Does it mean that the crons then won't run? It might help to clarify it a little. At least for me, I wasn't immediately sure what the contradiction would mean in practice.

Author: The cron will run, but the scheduled tasks won't be triggered. "Out-of-beat" schedules will be ignored. Thanks for raising this. Let me update the comment.

> This [blogpost](https://platform.sh/blog/of-cicadas-and-cron-jobs/) may help understanding the stakes and harmonazing the frequencies so all your scheduled tasks can be effectively triggered.

Reviewer: ```suggestion
This [blog post](https://platform.sh/blog/of-cicadas-and-cron-jobs/) may help you understand the stakes and harmonize the frequencies so all your scheduled tasks can be effectively triggered.
```

> You could also [configure a worker](../../../configuration/app/workers) relying on the `artisan schedule:work`.

Reviewer: ```suggestion
You could also [configure a worker](../../../configuration/app/workers.md) that relys on `artisan schedule:work`.
```

> [This command](https://laravel.com/docs/scheduling#running-the-scheduler-locally) will invoke the scheduler every minute. 

Reviewer: ```suggestion
To invoke the scheduler every minute, run [the following command](https://laravel.com/docs/scheduling#running-the-scheduler-locally):
```

```yaml {location=".platform.app.yaml"}
workers:
   queue:
       size: S
       commands:

>            start: |
>           php artisan schedule:work

Reviewer: ```suggestion
           start: php artisan schedule:work
```
